src/main/java/github/scarsz/discordsrv/DiscordSRV.java

```diff
@@ -1754,66 +1755,89 @@ public void processChatMessage(Player player, Component message, String channel,
-            discordMessage = DiscordUtil.convertMentionsFromNames(discordMessage, getMainGuild());
+            discordMessageContent = DiscordUtil.convertMentionsFromNames(discordMessageContent, getMainGuild());
         } else {
-            discordMessage = discordMessage.replace("@", "@\u200B"); // zero-width space
             discordMessageContent = discordMessageContent.replace("@", "@\u200B"); // zero-width space
         }
 
-        GameChatMessagePostProcessEvent postEvent = api.callEvent(new GameChatMessagePostProcessEvent(channel, discordMessage, player, preEvent.isCancelled()));
+        String processedMessage = discordMessageContent;
+
+        if(!webhookMessageDelivery) {
+            //If webhook delivery is not enable, we add all the placeholders
+            String username = player.getName();
+            if (!reserializer) username = DiscordUtil.escapeMarkdown(username);
+
+            String displayName = MessageUtil.strip(player.getDisplayName());
+
+            //Replace the internal placeholders in the message pattern
+            String discordMessagePattern = (hasGoodGroup
+                    ? LangUtil.Message.CHAT_TO_DISCORD.toString()
+                    : LangUtil.Message.CHAT_TO_DISCORD_NO_PRIMARY_GROUP.toString())
+                    .replace("%displayname%", DiscordUtil.escapeMarkdown(displayName))
+                    .replace("%displaynamenoescapes%", displayName)
+                    .replace("%username%", username)
+                    .replaceAll("%time%|%date%", TimeUtil.timeStamp())
+                    .replace("%channelname%", channel != null ? channel.substring(0, 1).toUpperCase() + channel.substring(1) : "")
+                    .replace("%primarygroup%", userPrimaryGroup)
+                    .replace("%usernamenoescapes%", MessageUtil.strip(player.getName()))
+                    .replace("%world%", player.getWorld().getName())
+                    .replace("%worldalias%", MessageUtil.strip(MultiverseCoreHook.getWorldAlias(player.getWorld().getName())));
+            //Replace the PAPI placeholders in the message pattern
+            discordMessagePattern = PlaceholderUtil.replacePlaceholdersToDiscord(discordMessagePattern, player);
+
+            //Reserialize the message pattern, in the case the placeholders added more color codes
+            if (reserializer) {
+                discordMessagePattern =
+                        MessageUtil.reserializeToDiscord(
+                                //Serialize into Components
+                                LegacyComponentSerializer.legacySection().deserialize(
+                                        //Replace $ with §
+                                        discordMessagePattern.replace(
+                                                LegacyComponentSerializer.AMPERSAND_CHAR,
+                                                LegacyComponentSerializer.SECTION_CHAR)
+                                ));
```

Comment:
This stuff doesn't need to be reserialized, only the message content does.

Reply:
Why not? PAPI placeholders could contain color codes in their results

Reply:
Because users (and the default value) expect to use Discord markdown in MinecraftToDiscord configuration options, reserializing escapes existing markdown characters.
 
Yes I agree that there might be some content that could do with being reserialized besides the message, and that is planned for the future, but right now DiscordSRV isn't really ready internally to do that.

Reply:
This still needs to be addressed as per the above

Reply:
2c45672f6cabc794abbe2dea991b1c5e9a9cdfcb